### PR TITLE
Optimize house numbers loading

### LINKED ISSUE #60: Optimise house number loading
House number loading is slow and requires too much memory, we need to optimise it.

### PR DESCRIPTION
**Pull request by pierDipi**
**New features and improvements**

- Optimize house numbers loading by avoiding associating
  house numbers to each street from the start.

**Related issue**

Closes #60

### CODE REVIEW

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (andr3a87) [inline src/main/resources/reference.conf:1]: it is necessary this file in src/main/resources?

File: src/main/scala/it/agilelab/bigdata/gis/domain/managers/IndexSet.scala [hunk 2]
```diff
@@ -1,8 +1,9 @@
 package it.agilelab.bigdata.gis.domain.managers
 
-import it.agilelab.bigdata.gis.domain.models.{OSMBoundary, OSMStreetAndHouseNumber}
+import it.agilelab.bigdata.gis.domain.models.{OSMBoundary, OSMHouseNumber, OSMStreetAndHouseNumber}
 import it.agilelab.bigdata.gis.domain.spatialList.GeometryList
 
 case class IndexSet(boundaries: GeometryList[OSMBoundary],
```
Comment (andr3a87) [inline src/main/scala/it/agilelab/bigdata/gis/domain/managers/IndexSet.scala:6]: [out of scope] what is the difference between regions and boundaries?

File: src/main/resources/reference.conf (hunk 1 above)
Comment (pierDipi) [reply src/main/resources/reference.conf:1]: No, we don't need it.

File: src/main/scala/it/agilelab/bigdata/gis/domain/managers/IndexSet.scala (hunk 2 above)
Comment (pierDipi) [reply src/main/scala/it/agilelab/bigdata/gis/domain/managers/IndexSet.scala:6]: idk, I think we don't need regions at all, tests pass without it, removed.

**Comment by pierDipi** [comment]
Europe
```
real    29m28.516s
user    289m3.061s
sys     6m12.288s
```

**Comment by pierDipi** [comment]
```
real    24m40.371s
user    271m15.639s
sys     5m42.206s
```

File: src/main/resources/reference.conf (hunk 1 above)
Comment (andr3a87) [reply src/main/resources/reference.conf:1]: ok, let's delete it or move it in test/resources

Comment (pierDipi) [reply src/main/resources/reference.conf]: Done!

**Review by andr3a87 — APPROVED** [review]

**Pull request merged**